applications: nrf5340_audio: Fix CIS and BIS for upmerge

alexsven opened the pull request:
- Added ext adv for BIS
- Changed callbacks for CIS
- Removed capabilities for CIS

Signed-off-by: Alexander Svensen <alexander.svensen@nordicsemi.no>

alexsven committed:
applications: nrf5340_audio: use unicast server callbacks

Upstream commit 9ecbee258742 ("Bluetooth: audio: capabilities: Remove
callbacks") removed audio capability ops callbacks in favor of unicast
server callbacks.

Fix CIS and BIS
- Added ext adv for BIS
- Changed callbacks for CIS
- Removed capabilities for CIS

Signed-off-by: Alexander Svensen <alexander.svensen@nordicsemi.no>
Signed-off-by: Tomasz Moń <tomasz.mon@nordicsemi.no>

desowin commented:
@rick1082 and @erikrobstad Could you please review?

desowin commented on applications/nrf5340_audio/src/bluetooth/Kconfig.defaults:
```diff
@@ -86,8 +86,7 @@ config BT_ASCS_ASE_SNK_COUNT
 config BT_ASCS_ASE_SRC_COUNT
-	default 1 if STREAM_BIDIRECTIONAL
-	default 0
+	default 1
```
Why?

alexsven replied:
Due to a bug in host, fixed in https://github.com/zephyrproject-rtos/zephyr/pull/52026

desowin replied:
If I cherry-pick the fix as `[nrf fromtree]` then this change can be removed?

alexsven replied:
yes :)

desowin replied:
Both commits from the fix are now part of upmerge. Please update this PR.

desowin commented on applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:
```diff
@@ -87,7 +87,7 @@ static bool bitrate_check(const struct bt_codec *codec)
 static bool adv_data_parse(struct bt_data *data, void *user_data)
 {
-	if (data->type == BT_DATA_NAME_COMPLETE && data->data_len == DEVICE_NAME_PEER_LEN) {
+	if (data->type == 0x30 && data->data_len) {
```
Where is this magic number from?

alexsven replied:
It's the broadcast name number defined here: https://www.bluetooth.com/wp-content/uploads/2022/11/assigned_numbers_release.pdf, it hasn't been defined in host yet, but there is a PR for it (https://github.com/zephyrproject-rtos/zephyr/pull/52069)

desowin replied:
If the PR is likely to be merged soon, then I don't see anything wrong with adding it as `[nrf fromtree]` and then using it here. Please update me on the PR.

rick1082 commented on applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c:
```diff
@@ -141,18 +143,109 @@ static struct bt_audio_stream_ops stream_ops = { .sent = stream_sent_cb,
+	net_buf_simple_add_le16(&ad_buf, BT_UUID_BROADCAST_AUDIO_VAL);
+	net_buf_simple_add_le24(&ad_buf, broadcast_id);
+
+	ext_ad[0].type = 0x30;
```
We should use macro to define this.

jhn-nordic committed:
applications: asset_tracker_v2: add CONFIG_NEWLIB_LIBC_NANO=y

Adding CONFIG_NEWLIB_LIBC_NANO=y to prj.conf to fit flash
memory with all configurations.

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

desowin commented:
@alexsven please just update the commit in your branch. It will show with conflicts here but I will handle it just fine. I put the not fully ready patch (this PR) into upstream branch mostly to have CI run on all patches I have received at once (as the patches are independent and cover different areas).

alexsven committed:
applications: nrf5340_audio: Set ASCS_ASE_SRC to 0 if not bidir

- Revert bugfix due to PR in upstream Zephyr

Signed-off-by: Alexander Svensen <alexander.svensen@nordicsemi.no>

desowin commented:
I should have probably created a separate branch, e.g. `upmerge-submissions` alongside `upmerge` branch. Then the PRs would go against `upmerge-submissions`. Then I could freely put things out in the `upmerge branch` (to trigger full CI) without confusing GitHub what is really going on. Doing so would allow doing actual per-patchset review without GitHub going crazy.
